### PR TITLE
AMPI: manage freeing of keyvals from Comm_free and Comm_free_keyval

### PR DESCRIPTION
**Pull request by stwhite91**

### CODE REVIEW

File: src/libs/ck-libs/ampi/ampi.C [hunk 1]
```diff
@@ -1731,10 +1731,18 @@ int ampiParent::setAttr(int context, vector<int>& keyvals, int keyval, void* att
...
   }
   keyvals.push_back(keyval);
-  kvlist[keyval]->incRefCount();
+  kvlist[keyval]->incRefCount(); // TODO: is this needed?
   return setUserKeyval(context, keyval, attribute_val);
 }
 
+int ampiParent::freeKeyval(int context, vector<int>& keyvals, int keyval) noexcept {
+  if (keyval >= 0 && keyval < kvlist.size()) {
+    kvlist[keyval]->decRefCount();
```
Comment (matthiasdiener) [inline src/libs/ck-libs/ampi/ampi.C]: This doesn't seem to actually free the keyval?

File: doc/charm++/manual.rst [hunk 2]
```diff
@@ -1083,8 +1083,8 @@ several fields:
        void someEntry(parameters2);
    };
 
-Note that A must have a *migration constructor*, which is typically
-empty:
+Note that A must have a *migration constructor* if it is to be migratable.
```
Comment (matthiasdiener) [inline doc/charm++/manual.rst]: These manual changes seem to be unrelated to the PR.

**Comment by evan-charmworks** [comment]
I've combined this commit with follow-on commits into my evan/romio_fixes branch.
